Add a popover close button, spell note popover

jazeee opened the pull request:
Ref Popovers #49

I have a popover example working:
![image](https://user-images.githubusercontent.com/5629800/211175987-586dac22-55c8-45c8-809b-557d1582c1bd.png)

![image](https://user-images.githubusercontent.com/5629800/211175985-49f97b8b-c6bf-4863-998d-2ebb49933cb2.png)

Also, added a close button.
Moving or dragging a popover is not an easy topic in Roll20. We don't really have that kind of control.

![image](https://user-images.githubusercontent.com/5629800/211176004-55b3a7e4-ee96-4f0c-9dad-2bedee40411a.png)

Converted to pug to allow HTML reuse.
![image](https://user-images.githubusercontent.com/5629800/211221773-be6da8ff-95ea-42cb-8599-4dcd1965f031.png)

Commit:
Add a popover close button, spell note popover

Ref Popovers #49

I have a popover example working:
![image](https://user-images.githubusercontent.com/5629800/211175987-586dac22-55c8-45c8-809b-557d1582c1bd.png)

![image](https://user-images.githubusercontent.com/5629800/211175985-49f97b8b-c6bf-4863-998d-2ebb49933cb2.png)

Also, added a close button.
Moving or dragging a popover is not an easy topic in Roll20.
We don't really have that kind of control.

![image](https://user-images.githubusercontent.com/5629800/211176004-55b3a7e4-ee96-4f0c-9dad-2bedee40411a.png)

jazeee commented on ui/components/popover/popover.scss:
```diff
@@ -4,6 +4,9 @@ button.popoverButton {
   border-radius: 100%;
   margin: 0;
+  &.smallerPopoverButton {
+    padding: 2px 6px !important;
```
We have a lot of global styles that are hard to override due to depth of specificity.
Ideally, we whittle those down and reduce the need for `!important`

jazeee commented on ui/spells/spells.html:
```diff
@@ -93,7 +93,22 @@
+        <div class="spellNotePopover popoverContainer">
+          <div class="popover">
+            <!-- Buttons in a popover break stacking order, and pop out of the repeating stack. -->
+            <button name="act_closepopoverspellnotes" class="closePopoverButton" type="action">X</button>
```
Turns out that `position: relative` breaks Roll20 and the button's action functionality.
This button can only act as a top level button, which is fine... We want to close all popovers when we click on this.

jazeee commented on ui/spells/spells.scss:
```diff
@@ -171,4 +171,16 @@
+    left: -40vw;
+    top: -40px;
+    label {
+      text-align: left;
```
Labels default to text-align right at a root style setup. 😢

jazeee commented on ui/spells/spells.scss:
```diff
@@ -171,4 +171,16 @@
+@include popover-control('spellNotePopover');
+div.spellNotePopover {
+  .popover {
+    width: 500px;
```
We can make the notes section as large as we want now...

jazeee commented on ui/scripts/inlineHtml/inlineScripts.js:
```diff
@@ -483,6 +483,29 @@ on('sheet:opened', () => setAttrs({ chosenPopover: 'false' }));
+
+on('sheet:opened', closeAllSpellNotes);
+
+on('clicked:repeating_arcane:togglepopoverspellnote', function () {
```
I need to follow up and apply this for innate and divine.

johnlamping commented:
This seems good.
For the future, it might be worth experimenting with a way to make the note appear in new space under the spell, when you click on the note icon and  not disappear when you click on a different one. That way, a player can keep open whatever notes they want.
(A lot of my career has been based on being able to see exactly you want, all at once, without having to click around.)

Commit:
Convert to pug

Commit:
Extract spells note popover pug

Commit:
Add notes popover to all spell types

Extract code into new pug file.
Adjust popover button padding.
Add an emoji icon.

Commit:
Add fixed placement popover option

jazeee commented:
I made a few style adjustments.
Next will be to add the backdrop. I have an idea on how to handle this.
